app/controllers/usersController.js

```diff
@@ -2,7 +2,37 @@
 
 const errors = require('../errors'),
   logger = require('../logger'),
-  User = require('../models').User;
+  User = require('../models').User,
+  bcrypt = require('bcrypt'),
+  userService = require('../services/userService'),
+  sessionManager = require('../services/sessionManager');
+
+exports.login = (req, res, next) => {
+  const userLogin = req.body
+    ? {
+        email: req.body.email,
+        password: req.body.password
+      }
+    : {};
+
+  userService.getByEmail(userLogin.email).then(u => {
+    if (u) {
+      bcrypt.compare(userLogin.password, u.password).then(isValid => {
+        if (isValid) {
+          const auth = sessionManager.encode({ email: u.email });
+          res.status(201);
+          res.set(sessionManager.HEADER_NAME, auth);
+          logger.info(`The user ${u.email} was logged in succesfull`);
+          res.send(u);
+        } else {
+          next(errors.databaseError);
```

Comment:
If the password comparison is falsey you are returning a databaseError, I think you wanted to send some other error.